src/xenia/app/emulator_window.cc

```diff
@@ -1071,10 +1094,248 @@ void EmulatorWindow::SetInitializingShaderStorage(bool initializing) {
+
+  switch (button_combination.function) {
+    case ButtonFunctions::ToggleFullscreen:
+      ToggleFullscreen();
+
+      // Extra Sleep
+      std::this_thread::sleep_for(delay);
+      break;
+    case ButtonFunctions::RunPreviouslyPlayedTitle:
+      RunPreviouslyPlayedTitle();
+      break;
+    case ButtonFunctions::ClearMemoryPageState:
+      ToggleGPUSetting(gpu_cvar::ClearMemoryPageState);
+
+      // Assume the user wants ClearCaches as well
+      if (cvars::d3d12_clear_memory_page_state) {
+        GpuClearCaches();
+      }
+
+      // Extra Sleep
+      std::this_thread::sleep_for(delay);
+      break;
+    case ButtonFunctions::ReadbackResolve:
+      ToggleGPUSetting(gpu_cvar::ReadbackResolve);
+
+      // Extra Sleep
+      std::this_thread::sleep_for(delay);
+      break;
+    case ButtonFunctions::CpuTimeScalarSetHalf:
+      CpuTimeScalarSetHalf();
+      break;
+    case ButtonFunctions::CpuTimeScalarSetDouble:
+      CpuTimeScalarSetDouble();
+      break;
+    case ButtonFunctions::CpuTimeScalarReset:
+      CpuTimeScalarReset();
+      break;
+    case ButtonFunctions::ToggleControllerVibration:
+      ToggleControllerVibration();
+
+      // Extra Sleep
+      std::this_thread::sleep_for(delay);
+      break;
+    case ButtonFunctions::CloseWindow:
+      window_->RequestClose();
+      break;
+    case ButtonFunctions::Unknown:
+    default:
+      break;
+  }
+
+  std::this_thread::sleep_for(delay);
+}
+
+void EmulatorWindow::GamepadHotKeys() {
+  X_INPUT_STATE state;
+
+  const std::chrono::milliseconds thread_delay(75);
+
+  auto input_sys = emulator_->input_system();
+
+  if (input_sys) {
+    // SDL and XInput both support the guide button
+    while (true) {
+      // Block scope surrounding input_lock used to release the lock
+      {
+        auto input_lock = input_sys->lock();
+
+        for (uint32_t user_index = 0; user_index < MAX_USERS; ++user_index) {
+          input_sys->GetState(user_index, &state);
+        }
+      }
+
+      ProcessControllerHotkey(state.gamepad.buttons);
+
+      std::this_thread::sleep_for(thread_delay);
+    }
+  }
+}
+
+void EmulatorWindow::ToggleGPUSetting(gpu_cvar value) {
+  switch (value) {
+    case gpu_cvar::ClearMemoryPageState:
+      D3D12SaveGPUSetting(D3D12GPUSetting::ClearMemoryPageState,
+                          !cvars::d3d12_clear_memory_page_state);
+      break;
+    case gpu_cvar::ReadbackResolve:
+      D3D12SaveGPUSetting(D3D12GPUSetting::ReadbackResolve,
+                          !cvars::d3d12_readback_resolve);
+      break;
+  }
+}
+
+// Determine if the Xbox Gamebar is enabled via the Windows registry
+bool EmulatorWindow::IsUseNexusForGameBarEnabled() {
+#ifdef _WIN32
+  const LPWSTR reg_path = L"SOFTWARE\\Microsoft\\GameBar";
+  const LPWSTR key = L"UseNexusForGameBarEnabled";
+
+  DWORD value = 0;
+  DWORD dataSize = sizeof(value);
+
+  RegGetValue(HKEY_CURRENT_USER, reg_path, key, RRF_RT_DWORD, nullptr, &value,
```

Comment:
I guess it would be nice to check if function failed and returned error code. I suspect that someone might not have it and it will cause confusion